plutotv.py

```diff
@@ -381,170 +409,217 @@ def fix(text):
+                    xml += '\n\t\t<episode-num system="pluto.tv.number">' + str(program_list[x]['episode_number']) + '</episode-num>' #episode number
+                    xml += '\n\t\t<episode-num system="pluto.tv.slug">' + program_list[x]['episode_slug'] + '</episode-num>' #episode number
+                    xml += '\n\t\t<episode-num system="pluto.tv.id">' + program_list[x]['episode_id'] + '</episode-num>' #episode number
+                
+                    xml += '\n\t\t<category lang="' + 'en' + '">' + program_list[x]['episode_genre'] + '</category>' #category
+                    xml += '\n\t\t<category lang="' + 'en' + '">' + program_list[x]['episode_series_type'] + '</category>' #category
+                
+                #content rating key
+                #print(program_list[x]['contentRating'])
+                if program_list[x]['episode_rating'] != '': #if not blank
+                    try:
+                        xml += '\n\t\t<rating system="' + rating_system[program_list[x]['episode_rating'].lower()] + '">' #rating system
+                        xml += '\n\t\t\t<value>' + program_list[x]['episode_rating'] + '</value>' #rating
+                        xml += '\n\t\t\t<icon src="' + rating_logo[program_list[x]['episode_rating'].lower()] + '" />' #rating logo from dictionary
+                        xml += '\n\t\t</rating>' #end rating key
+                    except KeyError:
+                        keyErrors_contentRating.append(program_list[x]['episode_rating'])
+                        errorDetails_contentRating.append('Title: ' + program_list[x]['episode_name'] + ', Channel: ' + program_list[x]['channelSlug'])
+                
+                if program_list[x]['episode_firstAired'] != '': #if first aired is not blank add the premier
+                    if timeOriginal != "": #if we have the originallyAvailableAt add it to the tag
+                        xml += '\n\t\t<previously-shown start="' + timeOriginal + ' ' + offset + '" />'
+                    else: #else don't add start time to the tag
+                        xml += '\n\t\t<previously-shown />'
+                else: #if program is premiere add the tag
+                    xml += '\n\t\t<premiere />'
+                
+                if program_list[x]['episode_liveBroadcast'] == True or program_list[x]['episode_series_type'] == 'live':
+                    xml += '\n\t\t<live />'
+                
+                #finish
+                xml += '\n\t</programme>'
+                
+            x += 1
+
+    if makeXML == True:
+        xml += '\n</tv>\n'
 
-    xml += '\n</tv>\n'
-    xml = xml.decode('unicode_escape').encode('utf-8')
-    print('xml is ready to write')
-    #print(xml)
+        #xml = xml.encode()
+        #xml = xml.decode('unicode_escape')
+        #xml = xml.encode().decode('unicode_escape')
+        print('xml is ready to write')
+        #print(xml)
```

Comment:
@nwithan8

If I uncomment line 603, the xml appears to print fine inside the command prompt window.

I cannot get it to write the file in line 608, unless I add the utf-8 encoding. Problem is it converts `&amp;` and other symbols back to `&`.

I want it to write the file exactly as I print it in line 603.

I am lost on this one... any idea?

Reply:
Change "w" in the open(filename) function to "wb" to write bytes data rather than string data. Then don't convert the bytes to a string before writing to the file